administration/src/mui-modules/application/primitive-inputs/DateForm.tsx

```diff
@@ -41,28 +50,46 @@ const DateForm: Form<State, ValidatedInput, AdditionalProps, Options> = {
     const [touched, setTouched] = useState(false)
     const { showAllErrors, disableAllInputs } = useContext(FormContext)
     const validationResult = DateForm.validate(state, options)
-
     const isInvalid = validationResult.type === 'error'
 
     return (
-      <TextField
-        variant='standard'
-        fullWidth
-        style={{ margin: '4px 0', minWidth }}
-        type='date'
-        label={label}
-        required
-        inputProps={{
-          max: options.maximumDate ? formatISO(options.maximumDate, { representation: 'date' }) : undefined,
-          min: '1900-01-01',
-        }}
+      <CustomDatePicker
+        value={parseDateFromState(state.value)}
         disabled={disableAllInputs}
-        error={touched && isInvalid}
-        value={state.value}
-        sx={{ '& input[value=""]:not(:focus)': { color: 'transparent' } }}
+        error={(showAllErrors || touched) && isInvalid}
+        label={label}
+        minDate={minDate}
+        maxDate={options.maximumDate}
         onBlur={() => setTouched(true)}
-        onChange={e => setState(() => ({ type: 'DateForm', value: e.target.value }))}
-        helperText={(showAllErrors || touched) && isInvalid ? validationResult.message : ''}
+        onChange={date => {
+          setState(() => ({
+            type: 'DateForm',
+            // Catch invalid dates: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date#the_epoch_timestamps_and_invalid_date
+            value: date && !Number.isNaN(date.valueOf()) ? date.toISOString() : '',
+          }))
+        }}
+        textFieldHelperText={(showAllErrors || touched) && isInvalid ? validationResult.message : undefined}
+        textFieldSlotProps={{
+          variant: 'filled',
+          style: { marginTop: '12px', minWidth },
+          sx: {
+            '.MuiPickersInputBase-root': {
+              backgroundColor: 'transparent',
+            },
+            '.MuiPickersSectionList-root': {
+              paddingLeft: 0,
+            },
+            '.MuiInputLabel-root': {
+              transform: 'translate(0px, 13px) scale(1)',
+            },
+            '.MuiInputLabel-shrink': {
+              transform: 'translate(0px, -1.5px) scale(0.75)',
+            },
+            '.MuiFormHelperText-root': {
+              marginLeft: 0,
+            },
+          },
```

Comment:
```suggestion
          variant: 'standard',
          style: { margin: '4px 0', minWidth },
          sx: {
            'input': { paddingTop: '4px' },
            '.MuiInputAdornment-root': { transform: 'translateX(-8px)' }
          }
```

I tried locally to use the standard variant, and it seems to work better than filled, with only some minor adjustments.

Feel free to ignore this comment though.

Reply:
Agreed, this is better.